src/fparser/two/tests/fortran2003/test_intrinsics.py

```diff
@@ -277,3 +292,55 @@ def test_shadowed_intrinsic(f2003_parser):
     table = tables.lookup("my_mod")
     sym = table.children[0].lookup("dot_product")
     assert sym.primitive_type == "real"
+
+
+def test_shadowed_intrinsic_import(f2003_parser):
```

Comment:
You could add some other cases here where the use statement is somewhere else e.g. in the subroutine and any other cases you can think of.

Reply:
I've made it a bit more complete. I'll add a test for a submodule as well.

Reply:
There was no submodule example but I expect that is either explained or dealt with later.